Makefile

```diff
@@ -1,53 +1,82 @@
 	$(error Cluster has not been started! Use 'make start' to start a cluster)
 endif
 
-.PHONY: _install-cluster-deps
-_install-cluster-deps:
-	scripts/install_cluster_deps.sh
+.PHONY: _prepare
+_prepare:
+	$(call LOG,Preparing dependencies)
+	.sandbox/prepare.sh
 
 .PHONY: start
-start: _install-cluster-deps  ## Start a local Flyte cluster
-	k3d cluster create -p "$(FLYTE_PROXY_PORT):30081" --no-lb --k3s-server-arg '--no-deploy=traefik' --k3s-server-arg '--no-deploy=servicelb' --kubeconfig-update-default=false $(FLYTE_CLUSTER_NAME)
-	k3d kubeconfig write $(FLYTE_CLUSTER_NAME)
-	kubectl --context $(FLYTE_CLUSTER_CONTEXT) apply -f https://raw.githubusercontent.com/flyteorg/flyte/master/deployment/sandbox/flyte_generated.yaml
-	kubectl --context $(FLYTE_CLUSTER_CONTEXT) wait --for=condition=available deployment/{datacatalog,flyteadmin,flyteconsole,flytepropeller} -n flyte --timeout=10m
+start: _prepare  ## Start a local Flyte cluster
+	$(call LOG,Starting sandboxed Kubernetes cluster)
+	docker run -d --rm --privileged --name $(FLYTE_CLUSTER_NAME) -e K3S_KUBECONFIG_OUTPUT=/config/kubeconfig -v /var/run -v $(PWD)/.sandbox/data/config:/config -p $(KUBERNETES_API_PORT):$(KUBERNETES_API_PORT) -p $(FLYTE_PROXY_PORT):30081 $(K3S_CLUSTER_IMAGE) --https-listen-port $(KUBERNETES_API_PORT) --no-deploy=traefik --no-deploy=servicelb --no-deploy=local-storage --no-deploy=metrics-server > /dev/null
+	timeout 600 sh -c "until kubectl cluster-info &> /dev/null; do sleep 1; done"
+
+	$(call LOG,Deploying Flyte)
+	# TODO switch to https://raw.githubusercontent.com/flyteorg/flyte/master/deployment/sandbox/flyte_generated.yaml
+	kubectl apply -f https://raw.githubusercontent.com/flyteorg/flyte/07734da0a902887678a7901114dbd96481aeecbc/deployment/sandbox/flyte_generated.yaml
+	kubectl wait --for=condition=available deployment/{datacatalog,flyteadmin,flyteconsole,flytepropeller} -n flyte --timeout=10m
+	$(call LOG,"Flyte deployment ready! Use 'make console' to open the Flyte console on your browser.")
 
 .PHONY: teardown
-teardown: _requires-active-cluster _install-cluster-deps  ## Teardown Flyte cluster
-	k3d cluster delete $(FLYTE_CLUSTER_NAME)
+teardown: _requires-active-cluster  ## Teardown Flyte cluster
+	$(call LOG,Tearing down)
+	docker rm -f -v $(FLYTE_CLUSTER_NAME) > /dev/null
 
 .PHONY: status
-status: _requires-active-cluster _install-cluster-deps  ## Show status of Flyte deployment
-	kubectl --context $(FLYTE_CLUSTER_CONTEXT) get pods -n flyte
+status: _requires-active-cluster  ## Show status of Flyte deployment
+	kubectl get pods -n flyte
 
 .PHONY: console
-console: _requires-active-cluster _install-cluster-deps  ## Open Flyte console
+console: _requires-active-cluster  ## Open Flyte console
 	open "http://localhost:$(FLYTE_PROXY_PORT)/console"
+
+.PHONY: register
+register: _requires-active-cluster  ## Register Flyte cookbook workflows
+	$(call LOG,Registering example workflows)
+	# TODO: Get this working!
```

Comment:
This doesn't actually work yet. But we should be able to get it working with a bit more work.

Reply:
Currently fails at `docker_push`, predictably. Will likely need to rework `cookbook/common/Makefile`.
```
The push refers to repository [docker.io/library/flytecookbook]
fa62d9a8d917: Preparing
25360a5dfbae: Preparing
9f0dda961f88: Preparing
9581f403e2d7: Preparing
c9e530290f15: Preparing
2e4eae82c3c0: Waiting
aef89e25dd79: Waiting
c70febab0a4b: Waiting
5f70bf18a086: Waiting
e21f2b576bf8: Waiting
2ce3ee9fabd8: Waiting
9b3434966ee1: Waiting
8711af342595: Waiting
9eb82f04c782: Waiting
denied: requested access to the resource is denied
make: *** [../common/Makefile:98: docker_push] Error 1
make: *** [register] Error 2
```

Reply:
I changed docker_push to not really push if the registry isn't set... can you rebase?

Reply:
```
Successfully serialized 98 flyte objects
/bin/sh: flyte-cli: not found
```

Would it make sense to run `flyte-cli register-files` from within the built image?

Reply:
ummmm @wild-endeavor @katrogan, should we? I know we want to support a scenario where you register from flytectl and all... but for this scenario would it make sense to register from within the container? keeping all dependencies intact ...

Reply:
The benefit would be that we don't have to worry about keeping the `flyte-cli` version in sync in the user's environment.